gpflow/training/natgrad_optimizer.py

```diff
@@ -0,0 +1,388 @@
+        self._gamma = gamma
+        self._natgrad_op = None
+
+    @property
+    def gamma(self):
+        return self._gamma
+
+    @property
+    def minimize_operation(self):
+        return self._natgrad_op
+
+    def minimize(self, model, var_list=None, session=None, feed_dict=None,
+                 maxiter=1000, anchor=True, **kwargs):
+        """
+        Minimizes objective function of the model.
+        Natural Gradient optimizer works with variational parameters only.
+        There are two supported ways of transformation for parameters:
+            - XiNat
+            - XiSqrtMeanVar
+        Custom transformations are also possible, they should implement
+        `XiTransform` interface.
+
+            :param model: GPflow model.
+            :param session: Tensorflow session where optimization will be run.
+            :param var_list: List of pair tuples of variational parameters or
+                triplet tuple with variational parameters and ξ transformation.
+                By default, all parameters goes through XiNat() transformation.
+                For example your `var_list` can look as,
+                ```
+                var_list = [
+                    (q_mu1, q_sqrt1),
+                    (q_mu2, q_sqrt2, XiSqrtMeanVar())
+                ]
+                ```
+            :param feed_dict: Feed dictionary of tensors passed to session run method.
+            :param maxiter: Number of run interation. Default value: 1000.
+            :param anchor: Synchronize updated parameters for a session with internal
+                parameter's values.
+            :param kwargs: Extra parameters passed to session run's method.
+        """
+
+        if model is None or not isinstance(model, Model):
+            raise ValueError('Unknown type passed for optimization.')
+
+        self._model = model
+        session = model.enquire_session(session)
+        opt = self.make_optimization(model, session=session, var_list=var_list, **kwargs)
+        with session.as_default():
+            for _i in range(maxiter):
+                opt()
+        if anchor:
+            model.anchor(session)
+    
+
+    def make_optimizer_tensor(self, model, session, var_list=None):
+        """
+        Make Tensorflow optimization tensor.
+        This method builds natural gradients optimization tensor and initializes all
+        necessary variables created by the optimizer.
+
+            :param model: GPflow model.
+            :param session: Tensorflow session.
+            :param var_list: List of tuples of variational parameters.
+            :return: Tensorflow natural gradient operation.
+        """
+        with session.as_default(), tf.name_scope(self.name):
+            # Create optimizer variables before initialization.
+            return self._build_natgrad_step_ops(model, *var_list)
+
+    def make_optimization(self, model, session=None, var_list=None, **kwargs):
+        """
```

Comment:
Optimization (typo)

Reply:
I don't see typo. Is that in description? Line below?

Reply:
Optimizatin -> missing 'o'

Reply:
Ah, it is under the block with your comment. In review mode github does not show me anything below.